src/server/svr_comm.c

```diff
@@ -0,0 +1,413 @@
+	}
+
+        tpp_mcast_close(mtfd_replyhello_psvr);
+        mtfd_replyhello_psvr = -1;
+}
+
+/**
+ * @brief send a node stat request to peer server
+ * 
+ */
+void
+send_nodestat(void)
+{
+	struct attrl *pat;
+	struct attrl *head;
+	struct attrl *tail;
+	struct attrl *tmp;
+	int mtfd = open_ps_mtfd();
+	int rc;
+	static int time_last_sent = 0;
+
+	/* Do not udate the cache too often */
+	if (time_now < time_last_sent + 2)
+		return;
+
+	time_last_sent = time_now;
+	
+	if (mtfd == -1)
+		return;
+
+	pat = calloc(1, sizeof(struct attrl));
+	pat->name = ATTR_NODE_Mom;
+	head = tail = pat;
+	pat = calloc(1, sizeof(struct attrl));
+	pat->name = ATTR_NODE_Port;
+	tail->next = pat;
+	tail = tail->next;
+	pat = calloc(1, sizeof(struct attrl));
+	pat->name = ATTR_server_inst_id;
+	tail->next = pat;
+	tail = tail->next;
+
+	rc =  PBSD_status_put(mtfd,
+		PBS_BATCH_StatusNode,
+		"", head, NULL, PROT_TPP, NULL);
+	if (rc)
+		close_streams(mtfd, rc);
+
+	for(pat = head; pat != NULL; pat = tmp) {
+		tmp = pat->next;
+		free(pat);
+	}
+	tpp_mcast_close(mtfd);
+	mtfd = -1;
+}
+
+/**
+ * @brief
+ * 		Input is coming from peer server over a TPP stream.
+ *
+ * @par
+ *		Read the stream to get a Peer-Server request.
+ *
+ * @param[in] stream  - TPP stream on which the request is arriving
+ * @param[in] version - Version of protocol.
+ *
+ * @return none
+ */
+void
+ps_request(int stream, int version)
+{
+        server_t        *psvr;
+        int             ret = 0;
+	int 		rc = 0;
+        int             command;
+        struct	sockaddr_in	*addr;
+        svrinfo_t *psvr_info;
+        pbs_list_head ru_head;
+
+        DBPRT(("%s: stream %d version %d\n", __func__, stream, version))
+	addr = tpp_getaddr(stream);
+	if (version != PS_PROTOCOL_VER) {
+		log_errf(-1, __func__, "protocol version %d unknown from %s",
+			version, netaddr(addr));
+		stream_eof(stream, 0, NULL);
+		return;
+	}
+	if (addr == NULL) {
+		log_err(-1, __func__, "Sender unknown");
+		stream_eof(stream, 0, NULL);
+		return;
+	}
+
+	command = disrsi(stream, &ret);
+	if (ret != DIS_SUCCESS)
+		goto badcon;
+
+        if (command == PS_CONNECT) {
+		if ((psvr = get_peersvr(addr)) == NULL)
+                        goto badcon;
+
+                log_eventf(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_INFO, __func__, 
+                                "Peer server connected from %s", netaddr(addr));
+                psvr_info = psvr->mi_data;
+		if (psvr_info->msr_stream >= 0 && psvr_info->msr_stream != stream) {
+			DBPRT(("%s: stream %d from %s:%d already open on %d\n",
+				__func__, stream, pmom->mi_host,
+				ntohs(addr->sin_port), psvr_info->msr_stream))
+			tpp_close(psvr_info->msr_stream);
+			tdelete2((u_long)psvr_info->msr_stream, 0ul, &streams);
+		}
+		/* we save this stream for future communications */
+		psvr_info->msr_stream = stream;
+		psvr_info->msr_state &= ~INUSE_NEEDS_HELLOSVR;
+                tinsert2((u_long)stream, 0ul, psvr, &streams);
```

Comment:
This file is full or alignment issues and very skimpy function headers. Please fix all of these